pkg/api/v1/clients/kube/resource_client.go

```diff
@@ -378,29 +411,42 @@ func (rc *ResourceClient) Watch(namespace string, opts clients.WatchOpts) (<-cha
 	// watch should open up with an initial read
 	cacheUpdated := rc.sharedCache.addWatch()
 
-	go func() {
+	go func(watchedNamespace string) {
 		defer rc.sharedCache.removeWatch(cacheUpdated)
 		defer close(resourcesChan)
 		defer close(errs)
 
+		// Perform an initial list operation
 		updateResourceList()
 
 		for {
 			select {
 			case <-time.After(opts.RefreshRate): // TODO(yuval-k): can we remove this? is the factory takes care of that...
```

Comment:
@yuval-k I would remove it. This does not really refresh the cache, but merely re-lists its content. The actual resync period (that forces the reflector to check whether the caches are out of sync) can now be specified on the `KubeResourceClientFactory`. Not sure if this is needed somewhere though. In case, the same behavior can be achieved by periodically calling the `List` method on the client itself. What do you think?

Reply:
the idea is that this helps "kick the tires" to make sure event loops get started immediately upon calling a watch. if you noticed this function isn't necessary, maybe because a watch event happens immediately anyway, we could remove it. but i think removing it might disturb some of our event loops (in particular i believe function discovery benefitted from this)

Reply:
I think you might be referring to line 420? I was talking about the `TODO` @yuval-k left at line 424 (the timeout clause in the `select` statement).